docs/premium-features/custom-types-and-actions.md

```diff
@@ -0,0 +1,145 @@
+
+In Amplication, you can create and manage both enums and DTOs using the Custom Types & Actions feature, allowing you to define the data contracts and constraints for your application's API endpoints.
+
+### Create DTO
+
+To create a new DTO, follow these steps:
+
+1. Click on the "Add DTO" button in the top right of the APIs tab.
+2. In the dropdown, click on "Add DTO".
+3. Provide a name for your DTO.
+4. Click the "Create DTO" button to finalize the creation process.
+
+![Create DTO](./assets/custom-types-and-actions/create-dto.png)
+
+### Edit, Disable, and Delete DTO
+
+To edit an existing DTO, click on the DTOs dropdown within its associated Module in the APIs tab. You can modify the DTO's name, description, and members.
+
+To disable an existing DTO, click on the toggle and move it into the disabled state. The dot in the left sidebar will now change colors into Red. You won't be able to use this DTO within the module.
+
+To delete a DTO entirely, click on the delete button next to the toggle. A confirmation popup will appear. Click on the delete button again to fully delete the DTO.
+
+:::note
+Be cautious when deleting DTOs, as it may impact other parts of your application that depend on them.
+:::
+
+![Edit, Disable, and Delete DTO](./assets/custom-types-and-actions/edit-disable-delete-dto.png)
+
+## Actions
+
+Actions represent the API endpoints that define the operations available in your application. They specify the HTTP methods, paths, input, and output data structures for each endpoint.
+
+Actions use DTOs to define their input and output data. Each Action can have a request DTO (for input) and a response DTO (for output). Amplication generates built-in Actions based on your application's entities, but you can also create custom Actions to implement specific functionality.
+
+### Create Action
+
+To create a new Action, follow these steps:
+
+1. Click on the "Add Action" button in the top right of the APIs tab.
+2. Provide a name for your new Action.
+3. Click the "Create Action" button to finalize the creation process.
+
+![Edit, Disable, and Delete DTO](./assets/custom-types-and-actions/create-action.png)
+
+### Enable and Disable Actions
+
+Amplication gives you a complete overview of all Actions associated with a specific Module.
+
+This is useful when you want to temporarily disable certain endpoints without completely removing them from your application.
+
+To enable or disable an Action, simply toggle the switch next to the Action's name in the APIs tab.
+
+![Enable and Disable Actions](./assets/custom-types-and-actions/all-actions.png)
+
+### Edit and Delete Action
+
+Similar to DTOs, you can edit, delete, move, and duplicate Actions within the APIs tab. When editing an Action, you can modify its HTTP method, path, input DTO, output DTO, and other settings as needed. Remember to save your changes when you're done.
+
```

Comment:
Same as the DTO comment- I think we need to better explain how to add / edit the Action itself- choosing input and output DTOs, defining the REST and GQL parameters, including the "Input Source" for REST (see below) etc. 
![image](https://github.com/amplication/docs/assets/149934977/4a4aefad-75d0-4799-bade-43f52b217088)